digdag-standards/src/main/java/io/digdag/standards/operator/PyOperatorFactory.java

```diff
@@ -124,6 +125,25 @@ public TaskResult runTask()
                 .build();
         }
 
+        @Override
+        public TaskResult cleanup(TaskRequest request)
+        {
+            final Path projectPath = workspace.getProjectPath(); // absolute
+            final CommandContext commandContext = buildCommandContext(projectPath);
+            final long attemptId = request.getAttemptId();
+            final long taskId = request.getTaskId();
+            Config state = TaskState.of(request).params();
+            if (state.has("commandStatus")) {
+                logger.debug(String.format("Starting cleanup: attemptId=%d, taskId=%d",  attemptId, taskId));
+                try {
+                    exec.cleanup(commandContext, state);
+                } catch (IOException ex) {
+                    throw Throwables.propagate(ex);
+                }
+            }
+            return TaskResult.empty(request);
```

Comment:
When is this line executed?

Reply:
This will never be executed. Since it expects to throw Exception, about to change the cleanup method to void.

Reply:
This line is executed when the state does not have "commandStatus".  e.g. Not a polling-based task. But now I feel it'd better change to void as return type.

Reply:
Updated. https://github.com/treasure-data/digdag/pull/1480/commits/36655d268db2c1cd567d31400573f8bd8429421c